x/claim/keeper/mission.go

```diff
@@ -57,29 +49,26 @@ func (k Keeper) GetAllMission(ctx sdk.Context) (list []types.Mission) {
 	return
 }
 
-// CompleteMission triggers the completion of the mission and distribute the claimable portion of airdrop to the user
-// the method fails if the mission has already been completed
-func (k Keeper) CompleteMission(ctx sdk.Context, missionID uint64, address string) error {
-	airdropSupply, found := k.GetAirdropSupply(ctx)
-	if !found {
-		return errors.Wrapf(types.ErrAirdropSupplyNotFound, "airdrop supply is not defined")
-	}
-
+// CompleteMission saves the completion of the mission
+func (k Keeper) CompleteMission(
+	ctx sdk.Context,
+	missionID uint64,
+	address string,
+) (claimed math.Int, err error) {
 	// retrieve mission
-	mission, found := k.GetMission(ctx, missionID)
-	if !found {
-		return errors.Wrapf(types.ErrMissionNotFound, "mission %d not found", missionID)
+	if _, found := k.GetMission(ctx, missionID); !found {
+		return claimed, errors.Wrapf(types.ErrMissionNotFound, "mission %d not found", missionID)
 	}
 
 	// retrieve claim record of the user
 	claimRecord, found := k.GetClaimRecord(ctx, address)
 	if !found {
-		return errors.Wrapf(types.ErrClaimRecordNotFound, "claim record not found for address %s", address)
+		return claimed, errors.Wrapf(types.ErrClaimRecordNotFound, "claim record not found for address %s", address)
```

Comment:
`ClaimRecord` are "per address" right ? I'm confused that we return an error if the `ClaimRecord` is not found. 

Does that mean all the `ClaimRecords` are created at genesis ? 
But then only addresses registered in the genesis can complete mission ?

Reply:
Yes, you are right. Only the addresses inside the `ClaimRecords` created in the genesis can complete and claim the airdrop. 

https://github.com/ignite/modules/blob/08b367dd974beb757836263f8d3960e3d8f02aa8/config.yml#L40-L60

Reply:
OK, so gathering all these addresses will be one of the jobs of the `airdrop` plugin right ?

Reply:
yes!! the `airdrop` plugin will automatically generate this for us